### ISSUE TITLE
Unspecified data type in image_dataset_from_directory

### ISSUE THREAD

**Issue by AngCamp** [keras-team/keras#15500]
PURL(s) with the issue: https://keras.io/api/preprocessing/image/#imagedatasetfromdirectory-function
There are issues with the following entries:

**labels:** Either "inferred" (labels are generated from the directory structure), None (no labels), or a list/tuple of integer labels of the same size as the number of image files found in the directory. Labels should be sorted according to the alphanumeric order of the image file paths (obtained via os.walk(directory) in Python).
**label_mode:** - 'int': means that the labels are encoded as integers (e.g. for sparse_categorical_crossentropy loss). - 'categorical' means that the labels are encoded as a categorical vector (e.g. for categorical_crossentropy loss). - 'binary' means that the labels (there can be only 2) are encoded as float32 scalars with values 0 or 1 (e.g. for binary_crossentropy). - None (no labels).

Description of issue (what needs changing):
The data types to provide **label** input are not clearly described and there are no examples of what would be an acceptable input.  Should the list contain ints or strings?  What about floats?  And then there is a reference to categorical vectors but no link to a description of this data type.  Again an example would make this clear.

It may not be desirable or feasible to reorganize a directory of images as they may not have been named in a consistent manner or you may be relying on meta-data to determine a desired class. Currently when I try to supply the function with a vector of images the function does not even recognize that there are files in the directory I am giving it, but when I move up in the directory and supply the file containing the images instead I get a Dataset object for training a single class.

The errors are not useful and confusing. It is not clear why the function does not recognize that there are image files in the folder in one case and not another. Examples would make implementing this use case much easier.

Furthermore this use case seems to be broken, although I may have simply provided the wrong input but becuase of the documentation I have no way of knowing if the function is broken or if I am using it wrong.  I have raised that issue here:
https://github.com/keras-team/keras/issues/15501

Please help.
